internal/pipeline/executor_test.go

```diff
@@ -128,3 +130,72 @@ spec:
 		}
 	}
 }
+
+func TestResolveSources(t *testing.T) {
+	// directory with only a Kptfile
+	emptyPkg, err := writePkg("")
+	if err != nil {
+		t.FailNow()
+	}
+	defer os.RemoveAll(emptyPkg.Path())
+
+	// package with subpackages
+	p, err := writePkg("")
+	defer os.RemoveAll(p.Path())
+	assert.NoError(t, err)
+
+	// subdirectories of dir
+	subp1, err := writePkg(p.Path())
+	assert.NoError(t, err)
+	subp2, err := writePkg(p.Path())
+	assert.NoError(t, err)
+	defer os.RemoveAll(subp1.Path())
+	defer os.RemoveAll(subp2.Path())
+
+	// sorting for testing purposes
+	subDirs := []string{subp1.Path(), subp2.Path()}
+	sort.Strings(subDirs)
+
+	tests := map[string]struct {
+		p			*pkg
+		expected 	[]string
+	}{
+		"package without pipeline": {
+			p: emptyPkg, // pipeline sources defaults to "./*"
+			expected: []string{emptyPkg.Path()},
+		},
+
+		"empty directory, current directory only": {
+			p: emptyPkg.withPipeline().withSources([]string{"."}),
+			expected: []string{emptyPkg.Path()},
+		},
+
+		"empty directory, all sources": {
+			p: emptyPkg.withPipeline().withSources([]string{"./*"}),
+			expected: []string{emptyPkg.Path()},
+		},
+
+		"directory with subpackages, current directory only": {
+			p: p.withPipeline().withSources([]string{"."}),
+			expected: []string{p.Path()},
+		},
+
+		"directory with subpackages, all sources": {
+			p: p.withPipeline().withSources([]string{"./*"}),
+			expected: append([]string{p.Path()}, subDirs...),
+		},
+	}
+
+	for testName, test := range tests {
```

Comment:
We should use subtests here and avoid including the name of the test in the error messages.

Reply:
done, thank you